### PR TITLE
Convert git tests to pytest

### PR DESCRIPTION
**Pull request by gerrod3**
Separated out from #1148

### CODE REVIEW

File: pulp_ansible/tests/functional/api/git/test_crud_remotes.py [hunk 1]
```diff
@@ -1,40 +1,50 @@
...
-            self.assertEqual(body[k], getattr(remote, k))
+from pulpcore.client.pulp_ansible import ApiException
+
+
+@pytest.mark.parallel
+def test_crud_git_remote(ansible_remote_git_api_client):
+    """Create, Read, Partial Update, and Delete an Ansible Git remote."""
+    # Create Git remote
+    body = gen_ansible_remote(url="https://github.com/geerlingguy/ansible-role-adminer.git")
+    remote = ansible_remote_git_api_client.create(body)
```
Comment (mdellweg) [inline pulp_ansible/tests/functional/api/git/test_crud_remotes.py:16]: This should be generate with cleanup. Even if the test set's out to remove it later. But if we fail in the middle we want the cleanup still.

File: pulp_ansible/tests/functional/conftest.py [hunk 2]
```diff
@@ -99,6 +114,18 @@ def _ansible_repo_factory(**kwargs):
     return _ansible_repo_factory
 
 
+@pytest.fixture
+def sync_repo_factory(ansible_repo_api_client):
```
Comment (mdellweg) [inline pulp_ansible/tests/functional/conftest.py:118]: Would it be useful if this factory created the repository and sync it and then return?
In the current state it looks like a sync_helper to me. Also be careful with the name. pytest put's all fixture names in the same scope.

Comment (gerrod3) [reply pulp_ansible/tests/functional/conftest.py]: Renamed to `ansible_sync_factory`. Also added auto repo creation if not supplied.

File: pulp_ansible/tests/functional/conftest.py [hunk 3]
```diff
@@ -99,6 +114,20 @@ def _ansible_repo_factory(**kwargs):
...
+    """A factory to perform a sync on an Ansible Repository and return its updated data."""
+
+    def _sync(ansible_repo=None, **kwargs):
+        body = AnsibleRepositorySyncURL(**kwargs)
+        if ansible_repo is None:
+            ansible_repo = ansible_repo_factory()
+        monitor_task(ansible_repo_api_client.sync(ansible_repo.pulp_href, body).task)
+        return ansible_repo_api_client.read(ansible_repo.pulp_href)
+
+    yield _sync
```
Comment (mdellweg) [inline pulp_ansible/tests/functional/conftest.py:128]: No need to make it a generator if we don't have cleanup code.
```suggestion
    return _sync
```

Comment (gerrod3) [reply pulp_ansible/tests/functional/conftest.py]: Would this apply to the other factories that just use `gen_object_with_cleanup` and have no other cleanup code as well?

Comment (mdellweg) [reply pulp_ansible/tests/functional/conftest.py]: yes, i think so.

File: pulp_ansible/tests/functional/api/git/test_crud_remotes.py [hunk 4]
```diff
@@ -1,40 +1,50 @@
...
+    update_body = {"url": "https://github.com/geerlingguy/ansible-role-ansible.git"}
+    remote_update = ansible_remote_git_api_client.partial_update(remote.pulp_href, update_body)
+    monitor_task(remote_update.task)
+    remote = ansible_remote_git_api_client.read(remote.pulp_href)
+    assert remote.url == update_body["url"]
+    assert remote.metadata_only is False
+
+    # Ensure Git remote doesn't have a download policy
+    with pytest.raises(AttributeError):
+        getattr(remote, "policy")
```
Comment (mdellweg) [inline pulp_ansible/tests/functional/api/git/test_crud_remotes.py:33]: How about `assert not hasattr()`

**Review by mdellweg — APPROVED** [review]

**Pull request merged**